In [1]:
! ~/.local/bin/aws s3 ls dev-rhassan/tony/

2020-04-11 20:17:10      62233 tmax001.png
2020-04-11 20:17:20      61915 tmax002.png
2020-04-11 20:17:31      56764 tmax003.png
2020-04-11 20:17:41      55980 tmax004.png
2020-04-11 20:17:53      61237 tmax005.png
2020-04-11 20:18:04      59641 tmax006.png
2020-04-11 20:18:14      58922 tmax007.png
2020-04-11 20:18:25      50959 tmax008.png
2020-04-11 20:18:35      51150 tmax009.png
2020-04-11 20:18:45      58277 tmax010.png
2020-04-11 20:18:56      48143 tmax011.png
2020-04-11 20:19:06      60303 tmax012.png
2020-04-11 20:19:17      59588 tmax013.png
2020-04-11 20:19:27      51251 tmax014.png
2020-04-11 20:19:37      50935 tmax015.png
2020-04-11 20:19:48      59388 tmax016.png
2020-04-11 20:19:58      59163 tmax017.png
2020-04-11 20:20:08      57463 tmax018.png
2020-04-11 20:20:18      55062 tmax019.png
2020-04-11 20:20:28      50216 tmax020.png
2020-04-11 20:20:39      47077 tmax021.png
2020-04-11 20:20:49      49580 tmax022.png
2020-04-11 20:20:59      56110 tmax023.png
2020-04-11 

2020-04-11 21:11:21      57360 tmax315.png
2020-04-11 21:11:31      57653 tmax316.png
2020-04-11 21:11:42      57838 tmax317.png
2020-04-11 21:11:52      58009 tmax318.png
2020-04-11 21:12:02      57244 tmax319.png
2020-04-11 21:12:12      57655 tmax320.png
2020-04-11 21:12:23      57009 tmax321.png
2020-04-11 21:12:34      58366 tmax322.png
2020-04-11 21:12:44      58804 tmax323.png
2020-04-11 21:12:54      58094 tmax324.png
2020-04-11 21:13:05      58604 tmax325.png
2020-04-11 21:13:15      58266 tmax326.png
2020-04-11 21:13:26      59622 tmax327.png
2020-04-11 21:13:36      59339 tmax328.png
2020-04-11 21:13:46      60192 tmax329.png
2020-04-11 21:13:56      59519 tmax330.png
2020-04-11 21:14:07      57996 tmax331.png
2020-04-11 21:14:17      57679 tmax332.png
2020-04-11 21:14:27      58512 tmax333.png
2020-04-11 21:14:38      59808 tmax334.png
2020-04-11 21:14:48      59783 tmax335.png
2020-04-11 21:14:58      59415 tmax336.png
2020-04-11 21:15:09      59344 tmax337.png
2020-04-11 

In [2]:
! ~/.local/bin/aws s3 ls dev-rhassan/tony/ | wc

    366    1464   15738


In [3]:
! ~/.local/bin/aws s3 rm s3://dev-rhassan/tony/ --recursive

delete: s3://dev-rhassan/tony/tmax001.png
delete: s3://dev-rhassan/tony/tmax002.png
delete: s3://dev-rhassan/tony/tmax004.png
delete: s3://dev-rhassan/tony/tmax015.png
delete: s3://dev-rhassan/tony/tmax003.png
delete: s3://dev-rhassan/tony/tmax008.png
delete: s3://dev-rhassan/tony/tmax009.png
delete: s3://dev-rhassan/tony/tmax013.png
delete: s3://dev-rhassan/tony/tmax019.png
delete: s3://dev-rhassan/tony/tmax006.png
delete: s3://dev-rhassan/tony/tmax020.png
delete: s3://dev-rhassan/tony/tmax017.png
delete: s3://dev-rhassan/tony/tmax023.png
delete: s3://dev-rhassan/tony/tmax012.png
delete: s3://dev-rhassan/tony/tmax024.png
delete: s3://dev-rhassan/tony/tmax005.png
delete: s3://dev-rhassan/tony/tmax025.png
delete: s3://dev-rhassan/tony/tmax021.png
delete: s3://dev-rhassan/tony/tmax010.png
delete: s3://dev-rhassan/tony/tmax022.png
delete: s3://dev-rhassan/tony/tmax007.png
delete: s3://dev-rhassan/tony/tmax033.png
delete: s3://dev-rhassan/tony/tmax029.png
delete: s3://dev-rhassan/tony/tmax

delete: s3://dev-rhassan/tony/tmax195.png
delete: s3://dev-rhassan/tony/tmax210.png
delete: s3://dev-rhassan/tony/tmax202.png
delete: s3://dev-rhassan/tony/tmax214.png
delete: s3://dev-rhassan/tony/tmax215.png
delete: s3://dev-rhassan/tony/tmax208.png
delete: s3://dev-rhassan/tony/tmax213.png
delete: s3://dev-rhassan/tony/tmax203.png
delete: s3://dev-rhassan/tony/tmax216.png
delete: s3://dev-rhassan/tony/tmax211.png
delete: s3://dev-rhassan/tony/tmax206.png
delete: s3://dev-rhassan/tony/tmax217.png
delete: s3://dev-rhassan/tony/tmax183.png
delete: s3://dev-rhassan/tony/tmax219.png
delete: s3://dev-rhassan/tony/tmax220.png
delete: s3://dev-rhassan/tony/tmax218.png
delete: s3://dev-rhassan/tony/tmax221.png
delete: s3://dev-rhassan/tony/tmax205.png
delete: s3://dev-rhassan/tony/tmax212.png
delete: s3://dev-rhassan/tony/tmax222.png
delete: s3://dev-rhassan/tony/tmax228.png
delete: s3://dev-rhassan/tony/tmax226.png
delete: s3://dev-rhassan/tony/tmax224.png
delete: s3://dev-rhassan/tony/tmax

In [4]:
! ~/.local/bin/aws s3 ls dev-rhassan/tony/

In [5]:
! ~/.local/bin/aws s3 ls dev-rhassan/

                           PRE TRR_test_cases/
                           PRE africa-scene-li/
                           PRE africa-scene-lists/
                           PRE newsslcert/
2020-04-10 15:00:03        315 get-cw-log-stream
2020-04-10 13:15:07     276255 lpgs_integration_testing_test_case_512_TC-512-10000-list.txt
2020-04-01 19:06:46      89930 tc_512_cloud_end_to_end.pdf


In [6]:
from playLib.bo_bucket import bo_get_bucket_list
bucket_name = 'ga-et-data'
prefix='Cloud_Veg_ET/Data/TMAX/'
lista=bo_get_bucket_list(bucket_name, prefix)

List count = 366


In [7]:
lista[0:8]

['Cloud_Veg_ET/Data/TMAX/tmax_001_gw.tif.npy',
 'Cloud_Veg_ET/Data/TMAX/tmax_002_gw.tif.npy',
 'Cloud_Veg_ET/Data/TMAX/tmax_003_gw.tif.npy',
 'Cloud_Veg_ET/Data/TMAX/tmax_004_gw.tif.npy',
 'Cloud_Veg_ET/Data/TMAX/tmax_005_gw.tif.npy',
 'Cloud_Veg_ET/Data/TMAX/tmax_006_gw.tif.npy',
 'Cloud_Veg_ET/Data/TMAX/tmax_007_gw.tif.npy',
 'Cloud_Veg_ET/Data/TMAX/tmax_008_gw.tif.npy']

In [8]:

# This is serial and VERY SLOW!
import os

bucket='ga-et-data'
for i in range(1):
    
    file_obj=lista[i]
    prefix = '/'.join(file_obj.split('/')[0:-1]) + '/'
    #print(prefix)
    filen = file_obj.split('/')[-1]
    #print(filen)
    argv = ','.join([bucket,prefix,filen])
    print(argv)
    cmd = 'python3 ./png-maker/png-maker.py ' + argv
    os.system(cmd) 

ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_001_gw.tif.npy


In [9]:
! ~/.local/bin/aws s3 ls dev-rhassan/tony/ | wc

      1       4      43


In [10]:
! ~/.local/bin/aws s3 ls dev-rhassan/tony/

2020-04-12 13:45:34      62233 tmax001.png


# Part 2 Queues with RabitMQ

In [11]:
 ! sudo systemctl status rabbitmq-server.service | grep Active

   Active: inactive (dead) since Sat 2020-04-11 01:19:25 UTC; 1 day 12h ago


In [12]:
# hello

In [13]:
%%bash
#systemctl --type=service
systemctl status rabbitmq-server.service |grep Active


   Active: inactive (dead) since Sat 2020-04-11 01:19:25 UTC; 1 day 12h ago


In [14]:
! (cd png; docker-compose up --scale qdispatch=3 -d)

png_tony_1 is up-to-date
Stopping and removing png_qdispatch_4 ... 
rabbitmq is up-to-date
Recreating png_qdispatch_1 ... _4 ... done
Recreating png_qdispatch_2 ... 
Recreating png_qdispatch_3 ... 
Recreating png_qdispatch_3
Recreating png_qdispatch_2
Recreating png_qdispatch_1
eating png_qdispatch_2 ... done

In [15]:
! echo sleeping...

sleeping...


In [16]:

%%bash
for i in {1..10}; do
    echo waiting... ;
    sleep 1 ;
done

waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...
waiting...


In [17]:
import os

from playLib.qbird import Qbird
from playLib.bo_bucket import bo_get_bucket_list


bucket_name = 'ga-et-data'
prefix='Cloud_Veg_ET/Data/TMAX/'
lista=bo_get_bucket_list(bucket_name, prefix)

q = Qbird('localhost','tonyq')
q.q_declare()

bucket='ga-et-data'
for file_obj in lista:
    
    prefix = '/'.join(file_obj.split('/')[0:-1]) + '/'
    #print(prefix)
    filen = file_obj.split('/')[-1]
    #print(filen)
    argv = ','.join([bucket,prefix,filen])
    #print(argv)
    cmd = 'png-maker.py ' + argv
    print(cmd) 
    msg=cmd
    q.q_publish(msg)

List count = 366
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_001_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_002_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_003_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_004_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_005_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_006_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_007_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_008_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_009_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_010_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_011_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_012_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_013_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_014_gw.tif.npy
png-maker.py ga-et-data,Cloud_V

png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_217_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_218_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_219_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_220_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_221_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_222_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_223_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_224_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_225_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_226_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_227_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_228_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_229_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_230_gw.tif.npy
png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,

In [18]:
! docker ps

CONTAINER ID        IMAGE                          COMMAND                  CREATED             STATUS              PORTS                                                                                        NAMES
f48801388ee8        tbutzer/q_spawner              "/usr/bin/python3 q_…"   14 seconds ago      Up 11 seconds                                                                                                    png_qdispatch_2
e62c3a275a8b        tbutzer/q_spawner              "/usr/bin/python3 q_…"   14 seconds ago      Up 12 seconds                                                                                                    png_qdispatch_3
f14b41923105        tbutzer/q_spawner              "/usr/bin/python3 q_…"   24 seconds ago      Up 22 seconds                                                                                                    png_qdispatch_1
6d764380d684        rabbitmq:3-management-alpine   "docker-entrypoint.s…"   18 hours ago        Up 18 hours   

In [19]:
! (cd png; docker-compose ps)

     Name                  Command             State             Ports          
--------------------------------------------------------------------------------
png_qdispatch_1   /usr/bin/python3             Up                               
                  q_spawner.py                                                  
png_qdispatch_2   /usr/bin/python3             Up                               
                  q_spawner.py                                                  
png_qdispatch_3   /usr/bin/python3             Up                               
                  q_spawner.py                                                  
png_tony_1        docker-entrypoint.sh redis   Up      6379/tcp                 
                  ...                                                           
rabbitmq          docker-entrypoint.sh rabbi   Up      15671/tcp,               
                  ...                                  0.0.0.0:15672->15672/tcp,
                            

In [20]:
! (cd png; docker-compose logs |grep qdispatch | tail -20)

Attaching to png_qdispatch_2, png_qdispatch_3, png_qdispatch_1, rabbitmq, png_tony_1
qdispatch_2  | 2020-04-12 13:45:59,178 - Q_SPAWNER - INFO - Logging Begins
qdispatch_2  | 2020-04-12 13:46:09,711 - Q_SPAWNER - INFO - png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_003_gw.tif.npy
qdispatch_2  | 2020-04-12 13:46:09,712 - Q_SPAWNER - INFO - ('I will spawn thee: ', 'png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_003_gw.tif.npy')
qdispatch_2  | 2020-04-12 13:46:09,712 - Q_SPAWNER - INFO - python3 ./png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_003_gw.tif.npy
qdispatch_2  | 2020-04-12 13:46:13,485 - PNG Builder - INFO - Logging Begins
qdispatch_2  | 2020-04-12 13:46:13,486 - PNG Builder - INFO - YAY!
qdispatch_3  | 2020-04-12 13:45:58,137 - Q_SPAWNER - INFO - Logging Begins
qdispatch_3  | 2020-04-12 13:46:09,712 - Q_SPAWNER - INFO - png-maker.py ga-et-data,Cloud_Veg_ET/Data/TMAX/,tmax_002_gw.tif.npy
qdispatch_3  | 2020-04-12 13:46:09,712 - Q_SPAWNER - INFO - ('I will spa

In [21]:
! date

Sun Apr 12 13:46:14 UTC 2020


In [22]:
! sudo docker exec rabbitmq rabbitmqctl status

Status of node rabbit@6d764380d684 ...
Runtime

OS PID: 331
OS: Linux
Uptime (seconds): 63882
RabbitMQ version: 3.8.3
Node name: rabbit@6d764380d684
Erlang configuration: Erlang/OTP 22 [erts-10.7.1] [source] [64-bit] [smp:2:2] [ds:2:2:10] [async-threads:64]
Erlang processes: 470 used, 1048576 limit
Scheduler run queue: 1
Cluster heartbeat timeout (net_ticktime): 60

Plugins

Enabled plugin file: /etc/rabbitmq/enabled_plugins
Enabled plugins:

 * rabbitmq_management
 * rabbitmq_web_dispatch
 * rabbitmq_management_agent
 * amqp_client
 * cowboy
 * cowlib

Data directory

Node data directory: /var/lib/rabbitmq/mnesia/rabbit@6d764380d684

Config files

 * /etc/rabbitmq/rabbitmq.conf

Log file(s)

 * <stdout>

Alarms

(none)

Memory

Calculation strategy: rss
Memory high watermark setting: 0.4 of available memory, computed to: 3.3124 gb
code: 0.0302 gb (29.72 %)
other_system: 0.024 gb (23.63 %)
other_proc: 0.0222 gb (21.82 %)
allocated_unused: 0.0159 gb (15.68 %)
other_ets: 0.0029 gb (2.85 

In [23]:
! sudo docker exec rabbitmq rabbitmqctl list_queues

Timeout: 60.0 seconds ...
Listing queues for vhost / ...
name	messages
tonyq	0


# Q them again

# Drain these messages using the admin gui port 

http://localhost:15672/
        
## ssh tunnel via 03<tab> 15672 mhonk

In [24]:
! sudo docker exec rabbitmq rabbitmqctl list_queues

Timeout: 60.0 seconds ...
Listing queues for vhost / ...
name	messages
tonyq	0


In [25]:
! sudo netstat -anlp |grep docker-proxy |grep LISTEN

tcp6       0      0 :::5672                 :::*                    LISTEN      21411/docker-proxy  
tcp6       0      0 :::15672                :::*                    LISTEN      21375/docker-proxy  


## Shut down the services with docker-compose

In [26]:
! date

Sun Apr 12 13:46:26 UTC 2020


In [27]:
# SHUT Er DOWN NOW!!!!!!!
#! (cd png; docker-compose down)